cmd/vicadmin/vicadm.go

```diff
@@ -426,12 +427,12 @@ func main() {
 	// If we're in an ESXi environment, then we need
 	// to extract the userid/password from UserPassword
 	if vchConfig.UserPassword != "" {
-		upw := strings.Split(vchConfig.UserPassword, ":")
-		if len(upw) == 2 {
-			vchConfig.Target.User = url.UserPassword(upw[0], upw[1])
-		} else {
-			vchConfig.Target.User = url.User(upw[0])
-		}
+		newurl, _ := url.Parse(fmt.Sprintf("%s://%s@%s%s",
```

Comment:
Don't have to rebuild url string, and parse it again.
You can use url.UserPassword to create a new user with password, and set back to existing url object.